packages/communication-react/review/communication-react.api.md

```diff
@@ -1470,8 +1472,11 @@ export interface FileUploadAdapter {
     registerFileUploads?: (fileUploads: ObservableFileUpload[]) => void;
 }
 
-// @beta (undocumented)
-export interface FileUploadEventEmitter {
+// @beta
+export class FileUploadEventEmitter {
```

Comment:
Do we expect contoso to create instances of this class? Or just of `ObservableFileUpload`?
If not we should keep this as an interface (of which we have an internal implementation that handles the events subscription)

Reply:
Oh maybe ignore this, if `ObservableFileUpload` is extending this we have to publicly export this as a class

Reply:
I guess to avoid exporting this as a class and keep as an interface it would have to be something like:
```
/** public */
interface FileUploadSubscriber {
  on()...
  off()...
}

/** private */
class FileUploadEventEmitter implements FileUploadSubscriber {
  private _emitter: EventEmitter
  ... impl ...
}

/** public */
class ObservableFileUpload implements FileUploadSubscriber {
  private fileUploadSubscriber: FileUploadSubscriber;

  // propagate to privately implemented FileUploadEventEmitter 
  on(event: 'uploadFailed', listener: UploadFailedListener) {
    this.fileUploadSubscriber.on(ev, listener);
  }
  etc.
}
```

Not sure if we feel strongly exporting classes. The reason to avoid is they are harder to maintain/easier to break if a customer ends up using the exported class - interfaces are much simpler

Reply:
So in your code sample, `ObservableFileUpload` will need to receive an instance of `FileUploadSubscriber` as a constructor parameter.
However, the expectation here is that Contoso will create a `new ObservableFileUpload` object for uploading files that are already uploaded to a cloud store.

Reply:
@JamesBurnside created an alternative PR
https://github.com/Azure/communication-ui-library/pull/1556
I think it will address all your comments.